build.sh

```diff
@@ -1,46 +1,141 @@
+    echo ""
+    echo "Description:"
+    echo "    This function will create a file \$DIR/korebuild-lock.txt. This lock file can be committed to source, but does not have to be."
+    echo "    When the lockfile is not present, KoreBuild will create one using latest available version from \$channel."
+    exit 2
+}
 
-if test ! -d $buildFolder; then
-    echo "Downloading KoreBuild from $koreBuildZip"
+#
+# Functions
+#
 
-    tempFolder="/tmp/KoreBuild-$(uuidgen)"
-    mkdir $tempFolder
+get_korebuild() {
 
-    localZipFile="$tempFolder/korebuild.zip"
+    local lock_file="$repo_path/korebuild-lock.txt"
+    if [ ! -f $lock_file ] || [ "$update" = true ]; then
+        __fetch "$tools_source/korebuild/channels/$channel/latest.txt" $lock_file
+    fi
+    local version=$(cat $lock_file | tail -1 | tr -d '[:space:]')
+    local korebuild_path="$DOTNET_HOME/buildtools/korebuild/$version"
+    if [ ! -d "$korebuild_path" ]; then
+        mkdir -p "$korebuild_path"
+        local remote_path="$tools_source/korebuild/artifacts/$version/korebuild.$version.zip"
 
-    retries=6
-    until (wget -O $localZipFile $koreBuildZip 2>/dev/null || curl -o $localZipFile --location $koreBuildZip 2>/dev/null)
-    do
-        echo "Failed to download '$koreBuildZip'"
-        if [ "$retries" -le 0 ]; then
-            exit 1
+        if ! __machine_has unzip; then
+            echo 'Missing required command: unzip'
+            return 1
         fi
-        retries=$((retries - 1))
-        echo "Waiting 10 seconds before retrying. Retries left: $retries"
-        sleep 10s
-    done
 
-    unzip -q -d $tempFolder $localZipFile
+        tmpfile="$(mktemp)"
+        echo "Downloading KoreBuild $version"
+        __fetch $remote_path $tmpfile
+        unzip -q -d "$korebuild_path" $tmpfile
+        rm $tmpfile || :
+    fi
+
+    source "$korebuild_path/KoreBuild.sh"
+}
 
-    mkdir $buildFolder
-    cp -r $tempFolder/**/build/** $buildFolder
+__machine_has() {
+    hash "$1" > /dev/null 2>&1
+    return $?
+}
 
-    chmod +x $buildFile
+__fetch() {
+    local remote_path=$1
+    local local_path=$2
+
+    if [[ "$remote_path" != 'http'* ]]; then
+        cp $remote_path $local_path
+        return 0
+    fi
 
-    # Cleanup
-    if test -d $tempFolder; then
-        rm -rf $tempFolder
+    failed=false
+    if __machine_has curl ; then
+        curl --retry 10 -sSL -f --create-dirs -o $local_path $remote_path || failed=true
+    elif __machine_has wget; then
+        wget --tries 10 -O $local_path $remote_path || failed=true
+    else
+        echo 'wget or curl is required to download assets' 1>&2
+        return 1
     fi
-fi
 
-$buildFile -r $repoFolder "$@"
+    if [ "$failed" = true ]; then
+        echo "Download failed: $remote_path" 1>&2
+        return 1
+    fi
+}
+
+#
+# main
+#
+
+while [[ $# > 0 ]]; do
+    case $1 in
+        -\?|-h|--help)
+            __usage
+            ;;
+        -c|--channel|-Channel)
```

Comment:
`-Channel` (and similar) are for equivalent syntax to PowerShell? Comment please :)